# Case Study on Testing of Hypothesis

A company started to invest in digital marketing as a new way of their product
promotions.For that they collected data and decided to carry out a study on it.

● The company wishes to clarify whether there is any increase in sales after
  stepping into digital marketing.
● The company needs to check whether there is any dependency between the
  features “Region” and “Manager”.
  
Help the company to carry out their study with the help of data provided.

In [2]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [3]:
#importing data
data=pd.read_csv('Sales_add.csv')

In [8]:
#data preprocessing
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


insight:
    1.There is no null values.
    2.Since the size  is < 30 we can do T-Test for the data.

In [10]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

insights:
    1.By comparing the above values, We can see an increase in Average sales
    (ie,Sales After investing in digital marketing is > Sales Before)

# ● whether there is any increase in sales after stepping into digital marketing?
    

In [ ]:
# n=22
Hypothesis:-
    H0[Null Hypothesis]-Sales before digital marketing and After digital marketing are same.
    H1[Alternate Hypothesis]- Sales after digital marketing is > the sales before digital marketing 

In [7]:
tscore, p_value = stats.ttest_ind(data['Sales_After_digital_add(in $)'],data['Sales_before_digital_add(in $)'],equal_var=True, alternative='greater')
print('pvalue is :', p_value)

pvalue is : 1.3071840034523225e-16


# result:
      

Since the p value(1.307) is less than 0.05 (Alpha value)
We  can reject the Null Hypothesis.
The Alternate Hypothesis is True. ie,
We can say that there is an increase in Sales after implementing digital marketing.

# ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

In [ ]:

#Hypothesis:-As the qn is to check the dependency we go for the chi_square Test-for Independence
   H0 [Null Hypothesis]- There is no dependency between the features “Region” and “Manager"
   H1 [Alternate Hypothesis]- There is significant dependency between the features “Region” 
       and “Manager"

In [5]:
from scipy.stats import chi2_contingency

In [6]:
con= pd.crosstab(data['Region'], data['Manager'])
con
c, p, dof, expected = chi2_contingency(con)
print('pValue is :',p)

pValue is : 0.5493991051158094


# result:

Since pValue(0.549) is higher than alpha (0.05), We can accept the Null hypothesis.
Which implies that, There is No Dependency between the features "Region" and "Manager".